### PR TITLE
[Mac] Fixed crash using Popover with `NSView`.

### PR DESCRIPTION
**Pull request by viniciusjarina**
* Added a parameter to `WrapWidget` to do not reparent the nativeView.
We can’t wrap and unwrap the NSView to be used with the Popover, because this will change the parent of the window when we unwrap it, invalidating the `NSView.Window`, making NSPopover throw a NSException, due the missing NSWindow inside the NSView.
* This add a option to do not reparent the native view when Wrap/Unwrap, and get the (untouched) native view back on `Show`.

Previous related PR: https://github.com/mono/xwt/pull/668

### CODE REVIEW

**Comment by chkn** [comment]
What is a case when you would want to pass `reparent: true`? Would it make more sense to just remove the reparenting behavior entirely?

**Comment by viniciusjarina** [comment]
@chkn I think not, all other parts of Xwt rely on this behavior, this is a corner case scenario where reparenting was causing the crash with `NSPopover` 😉

File: Xwt.XamMac/Xwt.Mac/PopoverBackend.cs [hunk 1]
```diff
@@ -203,7 +203,11 @@ public void DisableEvent (object eventId)
 		public void Show (Popover.Position orientation, Widget referenceWidget, Rectangle positionRect, Widget child)
 		{
 			var refBackend = Toolkit.GetBackend (referenceWidget) as IWidgetBackend;
-			NSView refView = (refBackend as ViewBackend)?.Widget;
+
+			NSView refView = (refBackend as EmbedNativeWidgetBackend)?.InnerView;
```
Comment (sevoku) [inline Xwt.XamMac/Xwt.Mac/PopoverBackend.cs]: Why is this required? Also it could break event handling, for events being attached after the widget has been added to the Xwt tree. I've got the feeling that we handle a product specific corner case here... and should make sure that we don't break other potential use cases.

File: Xwt.XamMac/Xwt.Mac/EmbedNativeWidgetBackend.cs [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
 	{
 		NSView innerView;
+		bool reparent;
 
 		public EmbedNativeWidgetBackend ()
 		{
 
 		}
 
+		public NSView InnerView {
```
Comment (sevoku) [inline Xwt.XamMac/Xwt.Mac/EmbedNativeWidgetBackend.cs]: Why is it necessary to acces the inner widget on XamMac backend level? Can't `Widget.Subview` be used to access the inner widget?

File: Xwt.Gtk/Xwt.GtkBackend/EmbeddedWidgetBackend.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ public EmbeddedWidgetBackend ()
 		{
 		}
 
-		public void SetContent (object nativeWidget)
+		public void SetContent (object nativeWidget, bool reparent)
```
Comment (sevoku) [inline Xwt.Gtk/Xwt.GtkBackend/EmbeddedWidgetBackend.cs]: The Gtk backend reparents the widget automatically by default: https://github.com/mono/xwt/blob/master/Xwt.Gtk/Xwt.GtkBackend/WidgetBackend.cs#L105
Since we introduce a new API here, I think we should throw an `NotImplementedException` in case `reparent: false`, or add the Gtk specific implementation for the new API.

File: Xwt.WPF/Xwt.WPFBackend/EmbedNativeWidgetBackend.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ namespace Xwt.WPFBackend
 {
 	class EmbedNativeWidgetBackend: WidgetBackend, IEmbeddedWidgetBackend
 	{
-		public void SetContent(object nativeWidget)
```
Comment (viniciusjarina) [inline Xwt.WPF/Xwt.WPFBackend/EmbedNativeWidgetBackend.cs]: 👍
